### PR TITLE
Eliminate issues from static analysis checks

### LINKED ISSUE #2: Fix static analysis issues
I added Psalm in #1 but there are a bunch of issues to be resolved.

### PR DESCRIPTION
**Pull request by asgrim**
Fixes #2 

Taking changes from @edigu in #3 and fixing remaining issues, adding Psalm to Travis build.

### CODE REVIEW

File: src/DriverFactory.php [hunk 1]
```diff
@@ -41,9 +36,8 @@ protected function createWithConfig(ContainerInterface $container, $configKey)
             $config['paths'] = [$config['paths']];
         }
 
-        if ($config['class'] === AnnotationDriver::class || is_subclass_of($config['class'], AnnotationDriver::class)) {
-            $this->registerAnnotationLoader();
```
Comment (asgrim) [inline src/DriverFactory.php]: Outstanding concern - original discussion:

https://github.com/Roave/psr-container-doctrine/pull/3#discussion_r376152219

Do we need to keep this @Ocramius ?

Comment (Ocramius) [reply src/DriverFactory.php]: I'd keep autoloader registration: it's a very basic concept, and we can actually enable it by default at all times (not even conditionally, tbh)

Comment (asgrim) [reply src/DriverFactory.php]: OK I'll add it back in :+1:

**Comment by asgrim** [comment]
@Ocramius reverted the `AnnotationRegistry::registerLoader` call - good to go?

File: src/DriverFactory.php [hunk 2]
```diff
@@ -89,4 +96,24 @@ protected function getDefaultConfig(string $configKey) : array
...
+    private function registerAnnotationLoader() : void
+    {
+        if (self::$isAnnotationLoaderRegistered) {
+            return;
+        }
+
+        /** @psalm-suppress DeprecatedMethod */
+        AnnotationRegistry::registerLoader(
+            /** @psalm-param class-string $className */
+            static function (string $className) {
```
Comment (Ocramius) [inline src/DriverFactory.php]: This can be `'class_exists'` as a string - no need for the closure

Comment (asgrim) [reply src/DriverFactory.php]: Done :+1:

**Pull request merged**